.github/workflows/main.yml

```diff
@@ -25,67 +28,88 @@ concurrency:
   group: ${{ github.head_ref || github.ref_name }}
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
-permissions: {} # No permissions by default on workflow level
+permissions: { } # No permissions by default on workflow level
 
 jobs:
   get-vars:
-    name: Get variables
-    env:
-      BUILD_ALL: ${{ github.event_name == 'workflow_dispatch' }}
+    name: Prepare build metadata
     runs-on: ubuntu-latest
     permissions:
       contents: read
     timeout-minutes: 10
     outputs:
-      image-tag: "${{ steps.image-tag.outputs.image-tag }}"
+      build_all: ${{ steps.set-vars.outputs.build_all }}
+      build_version: "${{ steps.prepare-build-version.outputs.version }}"
+      checkout_ref: ${{ steps.set-vars.outputs.checkout_ref }}
+      publish_binaries: ${{ steps.set-vars.outputs.publish_binaries }}
       components-list: ${{ steps.components-list.outputs.paths_list }}
       filtered-components-list: ${{ steps.filtered-components-list.outputs.filtered-list }}
       libs-list: ${{ steps.libs-list.outputs.paths_list }}
+
     steps:
+      - name: Prepare build variables
+        id: set-vars
+        env:
+          EVENT_NAME: ${{ github.event_name }}
+          GITHUB_REF: ${{ github.ref }}
+          DISPATCH_PUB: ${{ github.event.inputs.publish_binaries }}
```

Comment:
The `publish_binaries` input is only defined for `workflow_dispatch`. If the workflow is instead triggered by another event (`push`, `pull_request`, `merge_group`), what happens to this `DISPATCH_PUB` variable assignment? Does it raise an error or fallback to a null value?